### PR TITLE
chore: upgrade TS

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
The PR updates TypeScript versions across multiple packages to `5.5.2` and makes minor adjustments to package configurations.

### Detailed summary
- Updated TypeScript versions to `5.5.2` in various packages
- Adjusted `tsdk` setting in `vscode/settings.json`
- Modified object declarations in functions for clarity

> The following files were skipped due to too many changes: `packages/thirdweb/package.json`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
* **#3434** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3434?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/package.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
     "thirdweb": "workspace:*",
     "tiny-invariant": "^1.3.3",
     "use-debounce": "^10.0.1",
+    "viem": "^2.16.0",
```
Comment (jnsdls) [inline apps/dashboard/package.json:107]: do we explicitly need to add it here? why?

File: .vscode/settings.json [hunk 2]
```diff
@@ -11,5 +11,6 @@
   },
   "typescript.preferences.autoImportFileExcludePatterns": [
     "./packages/thirdweb/src/exports"
-  ]
+  ],
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (jnsdls) [inline .vscode/settings.json:15]: nice, makes sense

**Review by jnsdls — COMMENTED** [review]
please upgrade "typescript" everywhere u find it in dependencies, I believe it is also set in (all?) of the legacy packages and we need to update it there, too. (I don't want to fight mismatches)

File: apps/dashboard/package.json (hunk 1 above)
Comment (gregfromstl) [reply apps/dashboard/package.json]: Yep this was a mistake, only need to upgrade it where it's already installed

**Review by jnsdls — APPROVED** [review]
LGTM! Exited for how much this will let us delete explicit type definition wise.

**Pull request merged**